controllers/services/monitoring/monitoring_controller_actions.go

```diff
@@ -5,44 +5,109 @@ import (
 	"errors"
 	"fmt"
 
-	routev1 "github.com/openshift/api/route/v1"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
 
+	componentApi "github.com/opendatahub-io/opendatahub-operator/v2/apis/components/v1alpha1"
+	dscv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/datasciencecluster/v1"
 	serviceApi "github.com/opendatahub-io/opendatahub-operator/v2/apis/services/v1alpha1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/controllers/status"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster"
+	cr "github.com/opendatahub-io/opendatahub-operator/v2/pkg/componentsregistry"
 	odhtypes "github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/types"
-	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/metadata/labels"
-	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/resources"
+	odhdeploy "github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
 )
 
+// initialize handles all pre-deployment configurations.
 func initialize(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	log := logf.FromContext(ctx)
+	// Only handle manifests setup and initial configurations
+	platform := rr.Release.Name
+	switch platform {
+	case cluster.ManagedRhoai:
+		// Only set prometheus configmap path
+		rr.Manifests = []odhtypes.ManifestInfo{
+			{
+				Path:       odhdeploy.DefaultManifestPath,
+				ContextDir: "monitoring/prometheus/apps",
+			},
+		}
+
+	default:
+		log.Info("Monitoring enabled, won't apply changes in this mode", "cluster", platform)
+	}
+
 	return nil
 }
 
-func updateStatus(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
-	d, ok := rr.Instance.(*serviceApi.Monitoring)
-	if !ok {
-		return errors.New("instance is not of type *services.Monitoring")
+func updatePrometheusConfigMap(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	// Map component names to their rule prefixes
+	dscList := &dscv1.DataScienceClusterList{}
+	if err := rr.Client.List(ctx, dscList); err != nil {
+		return fmt.Errorf("failed to list DSC: %w", err)
+	}
+	if len(dscList.Items) == 0 {
+		return nil
+	}
+	dsc := &dscList.Items[0]
+	componentRules := map[string]string{
```

Comment:
nit: this map can probably be moved to a global var

Reply:
updated